process/block/preprocess/gasComputation.go

```diff
@@ -62,48 +70,78 @@ func NewGasComputation(
 	return g, nil
 }
 
-// Init method resets consumed and refunded gas structures
+// Init method inits provided, refunded and penalized gas structures
 func (gc *gasComputation) Init() {
 	gc.mutGasProvided.Lock()
 	gc.gasProvided = make(map[string]uint64)
+	gc.txHashesWithGasProvidedSinceLastReset = make([][]byte, 0)
 	gc.gasProvidedAsScheduled = make(map[string]uint64)
+	gc.txHashesWithGasProvidedAsScheduledSinceLastReset = make([][]byte, 0)
 	gc.mutGasProvided.Unlock()
 
 	gc.mutGasRefunded.Lock()
 	gc.gasRefunded = make(map[string]uint64)
+	gc.txHashesWithGasRefundedSinceLastReset = make([][]byte, 0)
 	gc.mutGasRefunded.Unlock()
 
 	gc.mutGasPenalized.Lock()
 	gc.gasPenalized = make(map[string]uint64)
+	gc.txHashesWithGasPenalizedSinceLastReset = make([][]byte, 0)
+	gc.mutGasPenalized.Unlock()
+}
+
+// Reset method resets tx hashes with gas provided, refunded and penalized since last reset
+func (gc *gasComputation) Reset() {
+	gc.mutGasProvided.Lock()
+	gc.txHashesWithGasProvidedSinceLastReset = make([][]byte, 0)
+	gc.txHashesWithGasProvidedAsScheduledSinceLastReset = make([][]byte, 0)
+	gc.mutGasProvided.Unlock()
+
+	gc.mutGasRefunded.Lock()
+	gc.txHashesWithGasRefundedSinceLastReset = make([][]byte, 0)
+	gc.mutGasRefunded.Unlock()
+
+	gc.mutGasPenalized.Lock()
+	gc.txHashesWithGasPenalizedSinceLastReset = make([][]byte, 0)
 	gc.mutGasPenalized.Unlock()
 }
 
 // SetGasProvided sets gas provided for a given hash
 func (gc *gasComputation) SetGasProvided(gasProvided uint64, hash []byte) {
 	gc.mutGasProvided.Lock()
 	gc.gasProvided[string(hash)] = gasProvided
+	gc.txHashesWithGasProvidedSinceLastReset = append(gc.txHashesWithGasProvidedSinceLastReset, hash)
 	gc.mutGasProvided.Unlock()
 }
 
 // SetGasProvidedAsScheduled sets gas provided as scheduled for a given hash
 func (gc *gasComputation) SetGasProvidedAsScheduled(gasProvided uint64, hash []byte) {
 	gc.mutGasProvided.Lock()
 	gc.gasProvidedAsScheduled[string(hash)] = gasProvided
+	gc.txHashesWithGasProvidedAsScheduledSinceLastReset = append(gc.txHashesWithGasProvidedAsScheduledSinceLastReset, hash)
```

Comment:
no need for temporary debug also for this line? 
```

//TODO: Remove it or set log level to Trace
	log.Debug("gasComputation.SetGasProvidedAsScheduled", "tx hash", hash, "gas gasProvided", gasProvided)
```

Reply:
Actually the problem happens only with the gas refunded and gas penalized